### PR TITLE
PyPI prep

### LINKED ISSUE #4: Add .gitignore
We need a .gitignore so that we don't accidentally commit .pyc files or any of the build directories.

### LINKED ISSUE #8: Add an entrypoint
Adding an entrypoint to the `setup.py` will make this easier to use from the command line.

To this we need to add something like

```python
    # ...
    entry_points={
        'console_scripts': ['edtf-validate'='ExtendedDataTimeFormat.valid_edtf:main']
    }
    # ...
```

as a keyword to the setup function. (We would have to wrap the `__name__ == '__main__'` branch in to a `main()` function for this configuration to work)

This will result in an auto-generated executable shell script that will be placed on the users PATH.

### PR DESCRIPTION
**Pull request by somexpert**
This PR is focused on getting this program ready for a PyPI release, which includes renaming, adding an entry point, and other slight modifications. This will fix #4 and fix #8.

### CODE REVIEW

**Comment by somexpert** [comment]
@ldko, @damonkelley, does these changes look sufficient for putting this on PyPI under version 1.0.0?

**Comment by somexpert** [comment]
@ldko, @damonkelley, this look good to tag and put up on PyPI?

File: README.md [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
+Valid EDTF provides validity testing against [EDTF](http://www.loc.gov/standards/datetime/pre-submission.html)
+levels 1-3. You might find it most useful for tasks involving date validation and comparison. Typical usage often looks like this:
 
 Import edtf into your own programs...
 
     #!/usr/bin/env python
-    from ExtendedDateTimeFormat.valid_edtf import isLevel1
+    from edtf_validate.valid_edtf import isLevel1
     if isLevel1(edtf_candidate):
         print "The date, %s, is level 1 edtf validated" % edtf_candidate
```
Comment (damonkelley) [inline README.md]: Can we unindent this and make it a fenced code block and add the python syntax?

File: README.md [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
 
 Or run it in the interpreter...
     
-    >>> import ExtendedDateTimeFormat
-    >>> from ExtendedDateTimeFormat.valid_edtf import is_valid
+    >>> import edtf_validate
+    >>> from edtf_validate.valid_edtf import is_valid
     >>> is_valid('1985')
     True
     >>> is_valid('Two Thousand Fourteen, The Year of Our Lord')
```
Comment (damonkelley) [inline README.md]: Further, can we combine this example and the previous?

I like the example code in the interpreter block, but I don't think we need the "python program" example **and** the "interpreter" version.

File: README.md [hunk 3]
```diff
@@ -22,10 +23,18 @@ Or run it in the interpreter...
 
 Or just straight from the commandline...
 
-    shabadoo@buntu:~/edtf$ python ExtendedDateTimeFormat/valid_edtf.py '1234'
-    1234	True
+    shabadoo@buntu:~/edtf$ edtf-validate 2015
+    2015	True
```
Comment (damonkelley) [inline README.md]: Let's make this generic and remove the hostname from the prompt.

```console
$ edtf-validate 2014
2015    True
```

Let's also convert this to a fenced code block and add the `console` syntax to it. (I used `console` on the example above)

File: README.md [hunk 4]
```diff
@@ -36,10 +45,19 @@ This program will:
...
+
+Installation
+------------
+
+The easiest way to install is through pip. To use pip to install edtf-validate, along with all the dependencies, use:
+
+    $ pip install edtf-validate
+
+
 Dependencies
```
Comment (damonkelley) [inline README.md]: What do you think we can get rid of this section completely?
1. pip **will** install the dependencies (not should)
2. All the deps are already listed in REQUIREMENTS.txt, the standard location for python package dependencies.

Comment (somexpert) [inline README.md]: Yeah it does seem unnecessary. I'm in agreement with removing it.

**Comment by damonkelley** [comment]
This is pretty nit-picky, but could we change `REQUIREMENTS.txt` to `requirements.txt`?

File: README.md [hunk 5]
```diff
@@ -1,31 +1,37 @@
-=========================
-Extended Date Time Format
+edtf-validate
 =========================
 
-Valid EDTF provides validity testing against EDTF levels 1-3. You might find it most useful for tasks involving date validation and comparison. Typical usage often looks like this:
+Valid EDTF provides validity testing against [EDTF](http://www.loc.gov/standards/datetime/pre-submission.html)
+levels 1-3. You might find it most useful for tasks involving date validation and comparison. Typical usage often looks like this:
 
 Import edtf into your own programs...
```
Comment (damonkelley) [inline README.md]: Let's get rid of this too. Doesn't make much sense now that we are only using the REPL snippet.

**Comment by damonkelley** [comment]
+1

File: edtf_validate/valid_edtf.py [hunk 6]
```diff
@@ -418,7 +418,7 @@ def is_valid(edtf_candidate):
         return False
 
 
-if __name__ == "__main__":
```
Comment (ldko) [inline edtf_validate/valid_edtf.py]: Since this was changed, thoughts on adding:

if __name__ == '__main__':
    main()

Comment (somexpert) [inline edtf_validate/valid_edtf.py]: Yeah that makes sense. Think that was originally my plan, but it slipped my mind. I'll add that in.

**Comment by ldko** [comment]
:+1:

**Pull request merged**